api/app/signals/apps/search/management/commands/elastic_index.py

```diff
@@ -1,19 +1,129 @@
+
+        if options['signal_id']:
+            self._index_signal(signal_id=int(options['signal_id']))
+        elif options['signal_ids']:
+            self._index_signals(signal_ids=list(map(int, options['signal_ids'].split(','))))
+        elif options['from_date'] or options['to_date']:
+            from_date = options['from_date'] or None
+            to_date = options['to_date'] or None
+            self._index_date_range(from_date, to_date)
+        elif options['_index_documents']:
+            self._index_documents()
+        else:
+            self.stdout.write('* No index option given')
+
+    def _clear_index(self):
+        self.stdout.write('* Clear the index')
+        if not self._dry_run:
             SignalDocument.clear_index()
-        SignalDocument.index_documents()
-        self.stdout.write('done!')
+            self._init_index()
+
+    def _init_index(self):
+        self.stdout.write('* Init the index')
+        if not self._dry_run:
+            SignalDocument.init()
+
+    def _index_documents(self):
+        self.stdout.write('* Index all Signals in bulk')
+        if not self._dry_run:
+            SignalDocument.index_documents()
+
+    def _index_signal(self, signal_id):
+        self.stdout.write(f'* Index Signal #{signal_id}')
+
+        try:
+            signal = Signal.objects.get(pk=signal_id)
+        except Signal.DoesNotExist:
+            self.stderr.write(f'* Signal #{signal_id} does not exists')
+            return None
+
+        if not self._dry_run:
+            SignalDocument.init()
+            signal_document = SignalDocument.create_document(signal)
+            signal_document.save()
+
+    def _index_signals(self, signal_ids):
+        self.stdout.write(f'* Indexing Signals: #{", #".join(map(str, signal_ids))}')
+
+        signal_qs = Signal.objects.filter(id__in=signal_ids).order_by('-updated_at')
+        signal_qs_ids = signal_qs.values_list('id', flat=True)
+        ids_diff = set(signal_ids) - set(signal_qs_ids)
+        if ids_diff:
+            self.stderr.write(f'* Signals not found #{", #".join(map(str, ids_diff))}')
+        else:
+            if not self._dry_run:
+                SignalDocument.index_documents(queryset=signal_qs)
+
+    def _index_date_range(self, from_date=None, to_date=None):
+        if not from_date and not to_date:
+            self.stderr.write('* At least a from date or a to date needs to be specified')
+            return
+
+        if from_date:
+            from_date = timezone.make_aware(timezone.datetime.strptime(from_date, '%Y-%m-%d'))
+        else:
+            from_date = timezone.now()
+        from_date = from_date.replace(hour=00, minute=00, second=00)  # Beginning of the given day
+
+        if to_date:
+            to_date = timezone.make_aware(timezone.datetime.strptime(to_date, '%Y-%m-%d'))
+        else:
+            to_date = timezone.now()
+        to_date = to_date.replace(hour=23, minute=59, second=59)  # End of the given day
```

Comment:
A better pattern is date + 1 day and time=00:00:00 for the end of the interval.

The SQL BETWEEN that the Django ORM range translates to, includes the boundaries of the interval. In this case that is no problem because a signal at the boundary of an interval is updated twice when you update consecutive days. Whilst the solution above will skip a second (and possibly signals).